spec/requests/saved_pages_request_spec.rb

```diff
@@ -0,0 +1,42 @@
+RSpec.describe "SavedPages", type: :request do
+  context "when receiving an unauthenticated request" do
+    it "returns unauthorised for GET /api/saved_pages" do
+      get saved_pages_path
+
+      expect(response).to have_http_status(:unauthorized)
+    end
+  end
+
+  context "when receiving an authenticated request" do
+    let(:oidc_client) { instance_double(OidcClient) }
+    let(:headers) { { "Content-Type" => "application/json", "GOVUK-Account-Session" => session_identifier } }
+    let(:session_identifier) { placeholder_govuk_account_session(user_id: user.sub) }
+    let(:user) { FactoryBot.create(:oidc_user) }
+
+    before do
+      stub_oidc_discovery
+
+      allow(OidcClient).to receive(:new).and_return(oidc_client)
+      allow(oidc_client).to receive(:tokens!).and_return({ access_token: "access-token", refresh_token: "refresh-token" })
+    end
+
+    describe "GET /api/saved_pages" do
```

Comment:
Should this be in line two and include both contexts ("when receiving an unauthenticated request" and "when receiving an authenticated request")?

Reply:
So as in why did i group these by auth/unauth _then_ the end points and not the other way around?

Mostly I figure the two fundimental states for everyhting here are `is this an authenticated request or not`?
No endpoint here is accessible without authentication, and all that is handled by the same auth code.

So I wanted to emphasise that at the top of the file, then with that out the way we can get down to tests (split out by endpoint) that describe the expeted functionality more.

Reply:
I see what you mean. I added the comment to the commit where only `GET /api/saved_pages` was tested so I mistakenly thought this file was only for one endpoint, hence my suggestion.

Either
```ruby
RSpec.describe "SavedPages", type: :request do
  context "when receiving an unauthenticated request" do
    describe "GET /api/saved_pages" do
    end

    describe "GET /api/saved_pages/:page_path" do
    end
  end
    
  context "when receiving an authenticated request" do
    describe "GET /api/saved_pages" do
    end

    describe "GET /api/saved_pages/:page_path" do
    end
  end
end
```
or
```ruby
RSpec.describe "SavedPages", type: :request do
  describe "GET /api/saved_pages" do
    context "when receiving an unauthenticated request" do
    end
    
    context "when receiving an authenticated request" do
    end
  end
  
  describe "GET /api/saved_pages/:page_path" do
    context "when receiving an unauthenticated request" do
    end
    
    context "when receiving an authenticated request" do
    end
  end
end
```
works for me.

Reply:
ah ace.

cool yeah, I think if we added an endpoint that could be accessed in a logged out state (though no idea what that'd be!) I'd reorder it then but this seemed to work for now?